### ISSUE TITLE
Make docstring headers in Sphinx bigger

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#17599]
Right now the headers in Sphinx for things like "examples", "parameters", and so on are just printed in bold. It would be better if they were larger, so they actually looked like headers. It would also be nice if they had anchors like real headers. 

CC @sympy/gsod

**Comment by moorepants** [comment]
Can you show an example of this? This seems like an unnecessary style change that we'd have to maintain in parallel to numpydoc. Why not submit upstream to numpydoc/rst/etc?

**Comment by asmeurer** [comment]
I don't think this is related to numpydoc, but rather the CSS in our Sphinx theme.

**Comment by asmeurer** [comment]
This is what the docstrings look like now. The headers look like bold text. ![image](https://user-images.githubusercontent.com/71486/65786892-e93a0500-e114-11e9-956f-9f71420f43c3.png)

Since with @lglattly's guide we will have a handful of preassigned headers people can use, it will be better if they render so that they look like actual headers in the HTML. This means making them bigger, and giving them anchors so that you can click on them and get a URL directly to them.

**Comment by moorepants** [comment]
Thanks, got it. When reading her doc I got the impression that this was a change to the numpydoc format. Making our css for the theme better is a good idea.

**Comment by asmeurer** [comment]
The headers are bigger now, but it would still be nice if we could add anchors to them like the top-level headers.
